### ISSUE TITLE
Json match fails on windows platform due to source path mismatch

### ISSUE THREAD

**Issue by amitkrout** [openshift/odo#2115]
[kind/bug]

## What versions of software are you using?
- Operating System: Windows10
- Output of `odo version`: master

## How did you run odo exactly?
```
------------------------------
odo app command tests when running app command without app parameter in directory that contains .odo config directory 
  should successfuly execute list, describe and delete along with machine readable output
  C:/Users/CDKQE/go/src/github.com/openshift/odo/tests/integration/cmd_app_test.go:68
Created dir: C:\Users\CDKQE\AppData\Local\Temp\142328437
Creating a new project: bvrylmcbus
Running odo.exe with args [odo project create bvrylmcbus -w -v4]
[odo]  -  Waiting for project to come up  ...
 V  Waiting for project to come up [583ms]
[odo]  V  Project 'bvrylmcbus' is ready for use
[odo]  V  New project created and now using project : bvrylmcbus
[odo] I0911 12:58:11.322563    3156 odo.go:70] Could not get the latest release information in time. Never mind, exiting gracefully :)
Current working dir: C:\Users\CDKQE\go\src\github.com\openshift\odo\tests\integration
Running odo.exe with args [odo component create nodejs nodejs --app app --project bvrylmcbus --context C:\Users\CDKQE\AppData\Local\Temp\142328437]
 V  Validating component [280ms]
[odo] Please use `odo push` command to create the component with source deployed
Running odo.exe with args [odo push --context C:\Users\CDKQE\AppData\Local\Temp\142328437]
[odo] Validation
 V  Checking component [485ms]
[odo]
[odo] Configuration changes
[odo]  V  Initializing component
 V  Creating component [1s]
[odo] 
[odo] Pushing to component nodejs of type local
 V  Checking files for pushing [8ms]
 V  Waiting for component to start [28s]
 V  Syncing files to the component [7s]
 V  Building component [9s]
[odo]  V  Changes successfully pushed to component
Setting current dir to: C:\Users\CDKQE\AppData\Local\Temp\142328437
Running odo.exe with args [odo app list]
[odo] The project 'bvrylmcbus' has the following applications:
[odo] NAME
[odo] app
Running odo.exe with args [odo list -o json]
[odo] {"kind":"List","apiVersion":"odo.openshift.io/v1alpha1","metadata":{},"items":[{"kind":"Component","apiVersion":"odo.openshift.io/v1alpha1","metadata":{"name":"nodejs","namespace":"bvrylmcbus","creationTimestamp":null},"spec":{"app":"app","type":"nodejs","source":"file:///./"},"status":{"state":"Pushed"}}]}
Setting current dir to: C:\Users\CDKQE\go\src\github.com\openshift\odo\tests\integration
Deleting project: bvrylmcbus
Running odo.exe with args [odo project delete bvrylmcbus -f]
[odo] This project contains the following applications, which will be deleted
[odo] Application app
[odo] This application has following components that will be deleted
[odo] component named nodejs
[odo] No services / could not get services
 V  Deleting project bvrylmcbus [23s]
[odo]  V  Deleted project : bvrylmcbus
Deleting dir: C:\Users\CDKQE\AppData\Local\Temp\142328437

+ Failure [80.474 seconds]
odo app command tests
C:/Users/CDKQE/go/src/github.com/openshift/odo/tests/integration/cmd_app_test.go:14
  when running app command without app parameter in directory that contains .odo config directory
  C:/Users/CDKQE/go/src/github.com/openshift/odo/tests/integration/cmd_app_test.go:62
    should successfuly execute list, describe and delete along with machine readable output [It]
    C:/Users/CDKQE/go/src/github.com/openshift/odo/tests/integration/cmd_app_test.go:68

    Expected
        <string>: {
          "kind": "List",
          "apiVersion": "odo.openshift.io/v1alpha1",
          "metadata": {},
          "items": [
            {
              "kind": "Component",
              "apiVersion": "odo.openshift.io/v1alpha1",
              "metadata": {
                "name": "nodejs",
                "creationTimestamp": null,
                "namespace": "bvrylmcbus"
              },
              "spec": {
                "type": "nodejs",
                "app": "app",
                "source": "file://./"
              },
              "status": {
                "state": "Pushed"
              }
            }
          ]
        }
    to match JSON of
        <string>: {
          "kind": "List",
          "apiVersion": "odo.openshift.io/v1alpha1",
          "metadata": {},
          "items": [
            {
              "kind": "Component",
              "apiVersion": "odo.openshift.io/v1alpha1",
              "metadata": {
                "name": "nodejs",
                "namespace": "bvrylmcbus",
                "creationTimestamp": null
              },
              "spec": {
                "app": "app",
                "type": "nodejs",
                "source": "file:///./"
              },
              "status": {
                "state": "Pushed"
              }
            }
          ]
        }

    
    first mismatched key: "items"[0]."spec"."source"

    C:/Users/CDKQE/go/src/github.com/openshift/odo/tests/integration/cmd_app_test.go:79
------------------------------
```

## Actual behavior
json match fails because machine readable output source path protocol is different on windows platform
windows:
```
"source": "file:///./"
```
Non windows:
```
"source": "file://./"
```

## Expected behavior
Should be handled in test script

**Comment by kadel** [comment]
Hmm, @amitkrout can you look if there is a solution for this in Ginkgo matcher library? Otherwise, we might need to write our own matcher to solve this :-(

**Comment by amitkrout** [comment]
Sure, i will checkout all possibilities

**Comment by girishramnani** [comment]
@amitkrout do you think we should close this issue?

**Comment by kadel** [comment]
> @kadel I have a created a quick fix #2118 for this issue.

We have quick a solution for this, not a nice one but it works, so I would close this.
/close

@amitkrout Feel free to reopen if you don't agree
